nixos/modules/services/networking/ferm.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ferm;
+
+  configFile = pkgs.stdenv.mkDerivation {
+    name = "ferm.conf";
+    text = cfg.config;
+    preferLocalBuild = true;
+    buildCommand = ''
+      echo -n "$text" > $out
+      ${cfg.package}/bin/ferm --noexec $out
+    '';
+  };
+in {
+  options = {
+    services.ferm = {
+      enable = mkEnableOption "Ferm Firewall";
+      config = mkOption {
+        description = "Verbatim ferm.conf configuration.";
+        default = "";
+        type = types.lines;
+      };
+      package = mkOption {
+        description = "The ferm package.";
+        type = types.package;
+        default = pkgs.ferm;
+        defaultText = "pkgs.ferm";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.firewall.enable = false;
```

Comment:
yeah. I agree this should be documented.
Automatic configuration could be done with ferm, probably even easier, because it
gives you data structures.
But currently I do not need this, as I want to build more complex stuff.
So I leave this feature to the future.